.github/workflows/haskell-ci.yml

```diff
@@ -0,0 +1,214 @@
+# This GitHub workflow config has been generated by a script via
+#
+#   haskell-ci 'github' 'cached-io.cabal'
+#
+# To regenerate the script (for example after adjusting tested-with) run
+#
+#   haskell-ci regenerate
+#
+# For more information, see https://github.com/haskell-CI/haskell-ci
+#
+# version: 0.14.3
+#
+# REGENDATA ("0.14.3",["github","cached-io.cabal"])
+#
+name: Haskell-CI
+on:
+  - push
+  - pull_request
```

Comment:
For public repos, let's use `pull_request_target`
https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#pull_request_target

Reply:
hmm reading through the docs again:
**This event runs in the context of the base of the pull request, rather than in the context of the merge commit, as the pull_request event does. This prevents execution of unsafe code from the head of the pull request that could alter your repository or steal any secrets you use in your workflow**. This event allows your workflow to do things like label or comment on pull requests from forks. **Avoid using this event if you need to build or run code from the pull request.**

do we need `pull_request_target` or just `pull_request`?

Reply:
This is autogenerated by the `haskell-ci` tool, so unless it's wildly wrong the cost of tweaking it will be much higher than just living with it.

Reply:
Builds need to be run in the context of the PR so I think `pull_request` is correct, but it should only be invokable by Bellroy staff so that a) we control our GHA billable time and b) we can ensure there is nothing malicious in the PR.

Reply:
Sure. Sounds good to me. Just wanted to make sure we understand what's happening.